### PR TITLE
Issue #97 fix. Absolute path in transfer

### PR DESCRIPTION
**Pull request by StyleT**
Fix for Issue #97. Now we can use absolute path in `transport.transfer`

### CODE REVIEW

**Comment by StyleT** [comment]
If I want to upload some file for me it's much more obvious to work with absolute paths:
```js
local.transfer([path.join(__dirname, '../somefile.sh'), '/log/somefile.txt'], '~/');
```

File: lib/transport/shell.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ Shell.prototype.transfer = function(files, remoteDir, options) {
                               (remote.username ? remote.username + '@' : ''),
                               remote.host, remoteDir);
 
-    var command = util.format('rsync --files-from %s %s --rsh="ssh -p%s%s" ./ %s',
+    var command = util.format('cat %s | xargs -n 50 -J % rsync %s --rsh="ssh -p%s%s" % %s',
```
Comment (pstadler) [inline lib/transport/shell.js]: A couple of questions:
- Why are you limiting this to 50 files?
- What are the percentage signs doing?
- Is this working with spaces and special characters in file names?
- Is this still working properly with relative paths?
- How does this perform when uploading hundreds of files?

File: README.md [hunk 2]
```diff
@@ -503,7 +503,9 @@ Copy a list of files to the current target's remote host(s) using
 flights.
 
 ```javascript
-var files = ['path/to/file1', 'path/to/file2'];
+var path = require('path');
+
+var files = [path.join(__dirname, '../path/to/file1'), '/path/to/file2'];
```
Comment (pstadler) [inline README.md]: This is generated from the source code.

File: test/test.transport.shell.js [hunk 3]
```diff
@@ -242,23 +243,9 @@ describe('transport/shell', function() {
         expect(WRITE_TEMP_FILE_STUB.calledOnce).to.be.true;
         expect(WRITE_TEMP_FILE_STUB.lastCall.args[0]).to.contain('/path/to/file');
         expect(WRITE_TEMP_FILE_STUB.lastCall.args[0]).to.contain('/path/to/another/file');
-        expect(EXEC_STUB.lastCall.args[0]).to.contain('--files-from /path/to/tmp/file');
-
-        testDone();
-      });
-    });
-
-    it('should upload an array of files', function(testDone) {
```
Comment (pstadler) [inline test/test.transport.shell.js]: Why did you remove this test?

**Comment by pstadler** [comment]
Thanks for your PR. Please check my comments.

File: lib/transport/shell.js (hunk 1 above)
Comment (StyleT) [inline lib/transport/shell.js]: 1. It's a chunk limit
1. See xargs doc http://charlesreid1.com/wiki/Xargs#-I_vs._-J
1. I hope :)
1. No, currently... But we can do some algorithm that will transform relative path into absolute. Question only one: what folder should be a "base"?
1. It will generate few rsync commands

File: test/test.transport.shell.js (hunk 3 above)
Comment (StyleT) [inline test/test.transport.shell.js]: It was a copypaste of the previous one with one extra line. I merged them.

File: README.md (hunk 2 above)
Comment (StyleT) [inline README.md]: Should I revert this changes?

Comment (pstadler) [inline README.md]: Change docs here: https://github.com/pstadler/flightplan/blob/master/lib/transport/index.js#L200
Rebuild with: `$ npm run docs`

File: lib/transport/shell.js (hunk 1 above)
Comment (pstadler) [inline lib/transport/shell.js]: 1. man xargs: "The current default value for number is 5000"
2. Please use %% (https://nodejs.org/api/util.html#util_util_format_format)
3. Can you verify this?
4. This is a requirement.

I think your changes break the current functionality where I can specify an array of files (in different or same directories) and the whole tree is rebuilt inside the remote's target directory.
